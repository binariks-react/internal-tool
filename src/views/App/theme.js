export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
