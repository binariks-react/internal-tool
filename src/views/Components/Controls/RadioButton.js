import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const RadioButtonWrapper = styled.label`
  display: flex;
  align-items: center;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
`;

const RadioButtonInput = styled.input`
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
`;

const RadioButtonIcon = styled.span`
  display: inline-block;
  box-sizing: border-box;
  color: #818994;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #fff;
  border: 1px solid #e1e5eb;
  transition: all 250ms cubic-bezier(0.27, 0.01, 0.38, 1.06);
  box-shadow: none;
  input:checked + & {
    border: 5px solid #007bff;
  }

  input[type='radio'][|checked][disabled] + & {
    background-color: #818994;
    border: 5px solid #e9ecef;
  }

  input:disabled + & {
    background-color: #e9ecef;
  }
`;

const RadioButtonSpan = styled.span`
  margin-left: 5px;
  color: #5a6169;

  input:disabled ~ &{
    color: #818994;
  }
`;

const RadioButton =	({ defaultChecked, disabled, title }) => (
  <RadioButtonWrapper {...{ disabled }}>
    <RadioButtonInput
      type="radio"
      {...{ defaultChecked, disabled }}
    />
    <RadioButtonIcon />
    <RadioButtonSpan>
      {title || 'radio'}
    </RadioButtonSpan>
  </RadioButtonWrapper>
);

const defaultProps = {
  defaultChecked: false,
  disabled: false,
  title: '',
};

const propTypes = {
  disabled: PropTypes.bool,
  defaultChecked: PropTypes.bool,
  title: PropTypes.string,
};

RadioButton.defaultProps = defaultProps;
RadioButton.propTypes = propTypes;

export default RadioButton;
