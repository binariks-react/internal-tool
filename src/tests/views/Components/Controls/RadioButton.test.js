import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount, render } from 'enzyme';
import theme from 'views/App/theme';
import RadioButton from 'views/Components/Controls/RadioButton';
import 'jest-styled-components';

describe('RadioButton', () => {
  it('default', () => {
    const radio = render(
      <ThemeProvider theme={theme}>
        <RadioButton />
      </ThemeProvider>
    );
    expect(radio).toMatchSnapshot();
  });
  it('defaultChecked', () => {
    const radio = render(
      <ThemeProvider theme={theme}>
        <RadioButton title="radio" />
      </ThemeProvider>
    );
    expect(radio).toMatchSnapshot();
  });
  it('disabled', () => {
    const radio = render(
      <ThemeProvider theme={theme}>
        <RadioButton title="radio" disabled />
      </ThemeProvider>
    );
    expect(radio).toMatchSnapshot();
  });
  it('onChange default', () => {
    const radio = mount(
      <ThemeProvider theme={theme}>
        <RadioButton />
      </ThemeProvider>
    );
    radio.find('input').simulate('change');
  });
});
