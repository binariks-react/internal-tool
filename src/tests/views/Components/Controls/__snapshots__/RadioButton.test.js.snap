// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  color: #818994;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #fff;
  border: 1px solid #e1e5eb;
  -webkit-transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  box-shadow: none;
}

input:checked + .c2 {
  border: 5px solid #007bff;
}

input[type='radio'][|checked][disabled] + .c2 {
  background-color: #818994;
  border: 5px solid #e9ecef;
}

input:disabled + .c2 {
  background-color: #e9ecef;
}

.c3 {
  margin-left: 5px;
  color: #5a6169;
}

input:disabled ~ .c3 {
  color: #818994;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="radio"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  >
    radio
  </span>
</label>
`;

exports[`RadioButton defaultChecked 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  color: #818994;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #fff;
  border: 1px solid #e1e5eb;
  -webkit-transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  box-shadow: none;
}

input:checked + .c2 {
  border: 5px solid #007bff;
}

input[type='radio'][|checked][disabled] + .c2 {
  background-color: #818994;
  border: 5px solid #e9ecef;
}

input:disabled + .c2 {
  background-color: #e9ecef;
}

.c3 {
  margin-left: 5px;
  color: #5a6169;
}

input:disabled ~ .c3 {
  color: #818994;
}

<label
  class="c0"
>
  <input
    checked=""
    class="c1"
    type="radio"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  >
    radio
  </span>
</label>
`;

exports[`RadioButton disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.c1 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  color: #818994;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #fff;
  border: 1px solid #e1e5eb;
  -webkit-transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  box-shadow: none;
}

input:checked + .c2 {
  border: 5px solid #007bff;
}

input[type='radio'][|checked][disabled] + .c2 {
  background-color: #818994;
  border: 5px solid #e9ecef;
}

input:disabled + .c2 {
  background-color: #e9ecef;
}

.c3 {
  margin-left: 5px;
  color: #5a6169;
}

input:disabled ~ .c3 {
  color: #818994;
}

<label
  class="c0"
  disabled=""
>
  <input
    class="c1"
    disabled=""
    type="radio"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  >
    radio
  </span>
</label>
`;

exports[`RadioButton disabled defaultChecked 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.c1 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  color: #818994;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #fff;
  border: 1px solid #e1e5eb;
  -webkit-transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  box-shadow: none;
}

input:checked + .c2 {
  border: 5px solid #007bff;
}

input[type='radio'][|checked][disabled] + .c2 {
  background-color: #818994;
  border: 5px solid #e9ecef;
}

input:disabled + .c2 {
  background-color: #e9ecef;
}

.c3 {
  margin-left: 5px;
  color: #5a6169;
}

input:disabled ~ .c3 {
  color: #818994;
}

<label
  class="c0"
  disabled=""
>
  <input
    checked=""
    class="c1"
    disabled=""
    type="radio"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  >
    radio
  </span>
</label>
`;
